File: 9amforlder/RetorfiteExxample9am/app/src/main/java/com/example/retorfiteexxample9am/RecyclerAdapter.java
package com.example.retorfiteexxample9am;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.MyViewholder> {


    List<PostModel> postList;
    Context context;

    public RecyclerAdapter(List<PostModel> postList, Context context) {
        this.postList = postList;
        this.context = context;
    }
    @NonNull
    @Override
    public RecyclerAdapter.MyViewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.post_item_layotu,parent,false);

        return new MyViewholder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerAdapter.MyViewholder holder, int position) {
        PostModel post = postList.get(position);
        holder.id.setText(post.getId());
        holder.title.setText(post.getTitle());
    }

    @Override
    public int getItemCount() {
        return postList.size();
    }

    public class MyViewholder extends RecyclerView.ViewHolder {
        TextView id,title;
        public MyViewholder(@NonNull View itemView) {
            super(itemView);
            id = itemView.findViewById(R.id.id);
            title = itemView.findViewById(R.id.title);
        }
    }
}

File: 4pmclass/TypesofAdapters/app/src/main/java/com/example/typesofadapters/apdatper/RecyclerAdapter.kt
package com.example.typesofadapters.apdatper

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.example.typesofadapters.R
import com.example.typesofadapters.modelclass.RecyclerMode

class RecyclerAdapter(var arraylist:MutableList<RecyclerMode>,var context: Context) : RecyclerView.Adapter<RecyclerAdapter.MyviewhOlder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerAdapter.MyviewhOlder {
       var view = LayoutInflater.from(parent.context).inflate(R.layout.layoutrecylerformat,parent,false)
        return MyviewhOlder(view)
    }



    override fun getItemCount(): Int {
        return arraylist.size
    }

    override fun onBindViewHolder(holder: RecyclerAdapter.MyviewhOlder, position: Int) {

        var recyclermod: RecyclerMode = arraylist.get(position)
      holder.imageView.setImageResource(recyclermod.imagerulr)
        holder.button.setOnClickListener {
            var nametext =holder.editText.text.toString()
            Toast.makeText(context," message => $nametext",Toast.LENGTH_SHORT).show()
        }
    }

    class MyviewhOlder(var itemview: View) : RecyclerView.ViewHolder(itemview)  {

         var imageView:ImageView
        var editText: EditText
         var button: Button

        init {
            imageView = itemview.findViewById(R.id.recyclrimage);
            editText = itemview.findViewById(R.id.namedittext)
            button = itemview.findViewById(R.id.simplebtn)
        }
    }
}